### ISSUE TITLE
Bug: Extracting to a canvas does not unpremultiply alpha

### ISSUE THREAD

**Issue by lunarraid** [pixijs/pixijs#10820]
### Current Behavior

Matt asked for an issue on this. Currently, calling extract.canvas returns a canvas whose pixel data contains premultiplied values, and due to the lossy nature of `putImageData`, there is no way to unpremultiply these correctly. The premultiplied image looks incorrect when extracting any image that contains opacity < 1.

### Expected Behavior

Extracting an image with opacity < 1 should return a correctly unpremultiplied canvas, that looks correct when overlaid as a normal DOM element.

### Steps to Reproduce

https://www.pixiplayground.com/#/edit/7-sDiAiwIPlyi0j3avx7q

Uncomment line 78 to see how it should look.

### Environment

- **`pixi.js` version**: *8*
- **Running Example**: https://www.pixiplayground.com/#/edit/7-sDiAiwIPlyi0j3avx7q

### Possible Solution

It looks like this was started but never finished: 

https://github.com/pixijs/pixijs/blob/fd7ddf6ef3aa4eb9fe8fc136585074821fc179cd/src/rendering/renderers/gl/texture/GlTextureSystem.ts#L405-L408

The playground also demonstrated the difference between calling this method and not.

**Comment by WishYouANiceDay** [comment]
Hi I am experiencing the same issue where I need unpremultiplied values. May I ask have you found any solution or workaround?

**Comment by lunarraid** [comment]
Yeah, you can find it in my 'Possible Solution'. You can read `getPixels` off of the extract, run `unpremultiplyAlpha` on it, then write that returned buffer to a canvas.

**Comment by WishYouANiceDay** [comment]
> Yeah, you can find it in my 'Possible Solution'. You can read `getPixels` off of the extract, run `unpremultiplyAlpha` on it, then write that returned buffer to a canvas.
> 
> (Don't try to run unpremultiplyAlpha on pixels that have already been written to a canvas, because you'll lose precision that way. The data just gets lost once written to a canvas.)

Oh I see. Thank you very much for the help and the example code.
